### PR TITLE
PUBDEV-4288: Set hadoop user from h2odriver

### PR DESCRIPTION
**Pull request by vitreuz**
This PR adds user manipulation control to the h2o hadoop drivers. This was primarily done for Steam (and any other H2O managing software) so as to avoid using root level privileges to impersonate users at the execution level.

Flags can be set to `-keytab <path> -principal <kerberos principal>` or `-run_as_user <valid user>` where the valid user must be a user in the hadoop environment.

(Note: using run as user implies the hadoop cluster doesn't have kerberos and is therefore already insecure)

### CODE REVIEW

File: h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java [hunk 1]
```diff
@@ -1020,6 +1033,18 @@ void validateArgs() {
       }
     }
 
+    if (principal != null || keytabPath != null) {
+      if (principal == null) {
+        error("principal requires a valid keytab path (use the '-keytab' option)");
```
Comment (mmalohlava) [inline h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java]: I think error message should complain about missing `principal`.

File: h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java [hunk 2]
```diff
@@ -1020,6 +1033,18 @@ void validateArgs() {
...
+      }
+      if (keytabPath == null) {
+        error("keytab requires a valid principal (use the '-principal' option)");
```
Comment (mmalohlava) [inline h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java]: The same as above.

File: settings.gradle [hunk 3]
```diff
@@ -54,9 +54,9 @@ if (System.getProperty("user.name").equals("jenkins")
 
   // Default hadoop build targets
   def allTargets = [
-          "cdh5.2", "cdh5.3", "cdh5.4", "cdh5.5", "cdh5.6", "cdh5.7", "cdh5.8",
-          "hdp2.1", "hdp2.2", "hdp2.3", "hdp2.4", "hdp2.5", "hdp2.6",
-          "mapr3.1", "mapr4.0", "mapr5.0", "mapr5.1","iop4.2"
+         "cdh5.2", "cdh5.3", "cdh5.4", "cdh5.5", "cdh5.6", "cdh5.7", "cdh5.8",
+         "hdp2.1", "hdp2.2", "hdp2.3", "hdp2.4", "hdp2.5", "hdp2.6",
+        "mapr4.0", "mapr5.0", "mapr5.1","iop4.2"
```
Comment (mmalohlava) [inline settings.gradle]: We can remove code as well if @tomkraljevic agrees.

**Comment by vitreuz** [comment]
Updated with @mmalohlava corrections

**Pull request merged**